activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,28 @@
+*   Strict loading using `:n_plus_one_only` does not eagerly load child associations.
+
+    Before:
+
+    ```ruby
+    person = Person.find(1)
+    person.strict_loading!(mode: :n_plus_one_only)
+    person.posts.first
+    # SELECT * FROM posts WHERE person_id = 1; -- non-deterministic order
+    ```
+    After:
+    ```ruby
+    person = Person.find(1)
+    person.strict_loading!(mode: :n_plus_one_only)
+    person.posts.first # this is 1+1, not N+1
+    # SELECT * FROM posts WHERE person_id = 1 ORDER BY id LIMIT 1;
+    ```
+    Strict loading in `:n_plus_one_only` mode is designed to prevent performance issues when
+    deeply traversing associations. It allows `Person.find(1).posts`, but _not_
+    `Person.find(1).posts.map(&:category)`. With this change, child associations are no
+    longer eagerly loaded, to match intended behavior and to prevent non-deterministic
+    order issues caused by calling methods like `first` or `last`.
+    Fixes #49473.
+    *Reid Lynch*
```

Comment:
We usually show the examples after the description of the change.
I think we can also skip the description of `:n_plus_one_only`:

```suggestion
    With this change, child associations are no
    longer eagerly loaded, to match intended behavior and to prevent non-deterministic
    order issues caused by calling methods like `first` or `last`.
    Fixes #49473.
    
    Before:

    ```ruby
    person = Person.find(1)
    person.strict_loading!(mode: :n_plus_one_only)
    person.posts.first
    # SELECT * FROM posts WHERE person_id = 1; -- non-deterministic order
    ```

    After:

    ```ruby
    person = Person.find(1)
    person.strict_loading!(mode: :n_plus_one_only)
    person.posts.first # this is 1+1, not N+1
    # SELECT * FROM posts WHERE person_id = 1 ORDER BY id LIMIT 1;
    ```

    *Reid Lynch*
```